server/queue_priv.php

```diff
@@ -3,12 +3,89 @@
 	require_once "$root/server/config.php";
 	$conn = $DB_CONN;
 
-	switch($_GET['action']) {
-		case 'display': displayAppointment($_GET['id']); break;
-		case 'cancel': cancelAppointment($_GET['id']); break;
+	// TODO: Make it so this is testable
+	require_once "$root/server/user.class.php";
+	$USER = new User();
+	if (!$USER->isLoggedIn()) {
+		header("Location: /unauthorized");
+		die();
+	}
+
+	switch($_REQUEST['action']) {
+		case 'display': displayAppointment($_REQUEST['id']); break;
+		case 'cancel': cancelAppointment($_REQUEST['id']); break;
+		case 'checkIn': checkIn($_REQUEST['time'], $_REQUEST['id']); break;
+		case 'completePaperwork': completePaperwork($_REQUEST['time'], $_REQUEST['id']); break;
+		case 'appointmentStart': appointmentStart($_REQUEST['time'], $_REQUEST['id']); break;
+		case 'appointmentComplete': appointmentComplete($_REQUEST['time'], $_REQUEST['id']); break;
+		case 'appointmentIncomplete': appointmentIncomplete($_REQUEST['explanation'], $_REQUEST['id']); break;
 		default: break;
 	}
 
+	function checkIn($time, $id) {
+		$stmt = $GLOBALS['conn']->prepare(
+			"INSERT INTO ServicedAppointment (timeIn, appointmentId)
+					VALUES (?, ?)"
+		);
+
+		$stmt->execute(array($time, $id));
+		$appointment = $stmt->fetchAll(PDO::FETCH_ASSOC);
+		echo json_encode($appointment);
+		$stmt = null;
+	}
+
+	function completePaperwork($time, $id) {
+		$stmt = $GLOBALS['conn']->prepare(
+			"UPDATE ServicedAppointment
+			SET ServicedAppointment.timeReturnedPapers = ?
+			WHERE ServicedAppointment.appointmentId = ?"
+		);
+
+		$stmt->execute(array($time, $id));
+		$appointment = $stmt->fetchAll(PDO::FETCH_ASSOC);
+		echo json_encode($appointment);
```

Comment:
Does this actually return anything? I don't think UPDATE statements actually return anything, except maybe the number of entries updated? I'm fairly certain you could probably remove:
```
$appointment = $stmt->fetchAll(PDO::FETCH_ASSOC);
echo json_encode($appointment);
```

I'm not 100% confident, but I think so... I didn't see the result being used on the front end at least. I think the thing that we probably should be returning here is like a success/error, and if it errors then we should display a popup or something on the front-end saying it failed, and asking them to try again. It also provides that ackowledgement to the front-end so that the front-end can adjust the stylings if it wasn't successful. I'm not too worried about the error handling back to the front-end right now. But it is something we should do eventually. I'll spin off a ticket for this.